bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/profiles/SystemProfileFactory.java

```diff
@@ -103,14 +107,32 @@ public SystemProfileFactory(final @Reference ChannelTypeRegistry channelTypeRegi
             return new SystemOffsetProfile(callback, context);
         } else if (RANGE.equals(profileTypeUID)) {
             return new SystemRangeStateProfile(callback, context);
+        } else if (BUTTON_TOGGLE_SWITCH.equals(profileTypeUID)) {
+            return new ToggleProfile<OnOffType>(callback, context, BUTTON_TOGGLE_SWITCH,
+                    DefaultSystemChannelTypeProvider.SYSTEM_BUTTON, OnOffType.ON, OnOffType.OFF,
+                    CommonTriggerEvents.SHORT_PRESSED);
```

Comment:
In the case of a raw button there are only `PRESSED` and `RELEASED`,  but here we have `SHORT_PRESSED`, `LONG_PRESSED` and `RELEASED`. Would it be possible to make it configurable if short or long events are used to toggle the state?

Reply:
It is already configurable. The event passed to the constructor is just the default event that is being used if the user does not configure it explicitely.